polar-core/src/resource_block.rs

```diff
@@ -46,68 +43,63 @@ enum ParsedDeclaration {
     Relations(Term),   // Dict<Symbol, Symbol>
 }
 
-fn validate_parsed_declaration((name, term): (Term, Term)) -> PolarResult<ParsedDeclaration> {
-    match (name.value().as_symbol()?.0.as_ref(), term.value()) {
+fn validate_parsed_declaration(
+    (name, term): (Term, Term),
+) -> Result<ParsedDeclaration, ValidationError> {
+    match (name.value().as_symbol().expect("parsed as symbol").0.as_ref(), term.value()) {
```

Comment:
Same as previous comment in `kb.rs` — we _know_ `name` is a `Term<Variable>` but we can't (yet) express that certainty via the type.